lib/ansible/modules/remote_management/manageiq/manageiq_tags.py

```diff
@@ -0,0 +1,248 @@
+
+    def post_tags(self, tags, action):
+        """ Post the tags for the resource using ManageIQ rest api
+        """
+        url = '{resource_url}/tags'.format(resource_url=self.resource_url)
+        try:
+            response = self.client.post(url, action=action, resources=tags)
+        except Exception as e:
+            msg = "Failed to {action} tag: {error}".format(
+                action=action,
+                error=e)
+            self.module.fail_json(msg=msg)
+
+        for result in response['results']:
+            if not result['success']:
+                msg = "Failed to {action}: {message}".format(
+                    action=action,
+                    message=result['message'])
+                self.module.fail_json(msg=msg)
+
+    def assign_or_unassign_tags(self, tags, action):
+        """ Assign or unassign the tag on a manageiq resource.
+
+        Returns:
+            Whether or not a change took place and a message describing the
+            operation executed.
+        """
+        tags_to_post = []
+        assigned_tags = self.query_resource_tags()
+        for tag in tags:
+            assigned = self.full_tag_name(tag) in assigned_tags
+
+            if assigned and action == 'unassign':
+                tags_to_post.append(tag)
+            elif (not assigned) and action == 'assign':
+                tags_to_post.append(tag)
+
+        if not tags_to_post:
+            return dict(
+                changed=False,
+                msg="Tags already {action}ed, nothing to do".format(action=action))
+        else:
+            self.post_tags(tags_to_post, action)
+            return dict(
+                changed=True,
+                msg="Successfully {action}ed tags".format(action=action))
+
+
+def main():
+    actions = {'present': 'assign', 'absent': 'unassign'}
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            manageiq_connection=dict(required=True, type='dict',
+                                     options=manageiq_argument_spec()),
+            tags=dict(required=True, type='list'),
+            resource_name=dict(required=True, type='str'),
+            resource_type=dict(required=True, type='str',
+                               choices=manageiq_entities().keys()),
+            state=dict(required=False, type='str',
+                       choices=['present', 'absent'], default='present'),
+        )
+    )
+
+    tags = module.params['tags']
+    resource_type_key = module.params['resource_type']
+    resource_name = module.params['resource_name']
+    state = module.params['state']
+
+    # get the action and resource type
+    action = actions[state]
+    resource_type = manageiq_entities()[resource_type_key]
+
+    # create the manageiq object
+    manageiq = ManageIQ(module)
+
+    # query resource id,
+    # fail if resource does not exist
+    resource_id = query_resource_id(manageiq, resource_type, resource_name)
+
+    # create the manageiq tag object
```

Comment:
IMHO these comments are pretty 1:1 to what the code does and can be dropped (only "fail if resource does not exist" above is helpful)
this one specifically is confusing, might sound like it's actually creating a tag in MIQ, while it's just instantiating our class...